addOns/llm/src/main/java/org/zaproxy/addon/llm/ExtensionLlm.java

```diff
@@ -0,0 +1,187 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.llm;
+
+import java.awt.CardLayout;
+import java.awt.Font;
+import java.awt.event.KeyEvent;
+import java.io.File;
+import java.nio.file.Files;
+import javax.swing.ImageIcon;
+import javax.swing.JTextPane;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.SessionChangedListener;
+import org.parosproxy.paros.model.Session;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.addon.llm.ui.LlmReviewAlertMenu;
+import org.zaproxy.addon.llm.ui.settings.LlmOptionsPanel;
+import org.zaproxy.addon.llm.ui.settings.LlmOptionsParam;
+import org.zaproxy.addon.llm.ui.ImportDialog;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+/**
+ * An example ZAP extension which adds a top level menu item, a pop up menu item and a status panel.
+ *
+ * <p>{@link ExtensionAdaptor} classes are the main entry point for adding/loading functionalities
+ * provided by the add-ons.
+ *
+ * @see #hook(ExtensionHook)
+ */
+public class ExtensionLlm extends ExtensionAdaptor {
+
+    // The name is public so that other extensions can access it
+    public static final String NAME = "ExtensionLlm";
+
+    // The i18n prefix, by default the package name - defined in one place to make it easier
+    // to copy and change this example
+    protected static final String PREFIX = "llm";
+
+    /**
+     * Relative path (from add-on package) to load add-on resources.
+     *
+     * @see Class#getResource(String)
+     */
+    private static final String RESOURCES = "resources";
+
+    private ZapMenuItem menuExample;
+    private AbstractPanel statusPanel;
+    private ImportDialog importDialog;
+    private LlmReviewAlertMenu llmReviewAlertMenu;
+    private LlmOptionsParam llmOptionsParam;
+    private LlmOptionsPanel llmOptionsPanel;
+    private static final String[] ROOT = {};
+
+    private LlmAPI api;
+
+    private static final Logger LOGGER = LogManager.getLogger(ExtensionLlm.class);
+
+    public ExtensionLlm() {
+        super(NAME);
+        setI18nPrefix(PREFIX);
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+        this.llmOptionsParam = new LlmOptionsParam();
+
+
+        if (hasView()) {
+            extensionHook.getHookMenu().addImportMenuItem(getMenuLLM());
+            extensionHook.getHookMenu().addPopupMenuItem(getCheckLlmMenu());
+            extensionHook.addOptionsParamSet(getOptionsParam());
+            // change to a message
```

Comment:
?

Reply:
can you please clarify what should be done ?

Reply:
Deal with the comment, or remove it.